prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -668,10 +668,12 @@ changesUponFlattening = \doc -> case doc of
 
 
 
--- | @('flatAlt' x fallback)@ renders as @x@ by default, but falls back to
--- @fallback@ when 'group'ed. Since the layout algorithms rely on 'group' having
--- an effect of shortening the width of the contained text, careless usage of
--- 'flatAlt' with wide fallbacks might lead to unappealingly long lines.
+-- | By default, @('flatAlt' x y)@ renders as @x@. However when 'group'ed,
+-- @y@ will be preferred, with @x@ as the fallback for the case where @y@
+-- doesn't fit.
```

Comment:
I think I should also mention that `group` will (attempt to) flatten `y`.

Reply:
…and that `y` will never be chosen if it's unflattenable! (https://github.com/quchen/prettyprinter/issues/152)